views/js/runner/plugins/content/responsiveness/collapser.js

```diff
@@ -75,28 +91,115 @@ define([
          * Installs the plugin (called when the runner bind the plugin)
          */
         init: function init() {
-            var testRunner = this.getTestRunner();
-            var testData = testRunner.getTestData() || {};
-            var testConfig = testData.config || {};
-            var pluginsConfig = testConfig.plugins || {};
-            var config = _.defaults(pluginsConfig.collapser || {}, defaults);
+            var testRunner = this.getTestRunner(),
+                testData = testRunner.getTestData() || {},
+                testConfig = testData.config || {},
+                pluginsConfig = testConfig.plugins || {},
+                config = _.defaults(pluginsConfig.collapser || {}, defaults),
+                collapseCls = config.hover ? noLabelHoverCls : noLabelCls;
+
             var areaBroker = testRunner.getAreaBroker();
-            var $actionsBar = areaBroker.getArea('actionsBar');
-            var $toolbox = areaBroker.getToolboxArea();
-            var $navigation = areaBroker.getNavigationArea();
-            var collapseCls = config.hover ? noLabelHoverCls : noLabelCls;
 
+            var $actionsBar = areaBroker.getArea('actionsBar'),
+                $toolbox = areaBroker.getToolboxArea(),
+                $navigation = areaBroker.getNavigationArea();
 
-            function getAvailableWidth() {
-                return $actionsBar.width();
+            var allCollapsibles,
+                availableWidth,
+                previousAvailableWidth,
+                totalExtraWidth;
+
+            /**
+             * Get a reference of all collapsibles
+             */
+            function buildCollapsiblesList() {
+                // use the given order to build the collapsibles list
+                if (shouldCollapseInOrder()) {
+                    allCollapsibles = config.collapseOrder
+                        .map(function(selector) {
+                            var $elements = $(selector).not('.' + labelHiddenCls); // some buttons are collapsed by configuration: we should leave them alone
+                            var extraWidth = 0;
+
+                            if ($elements.length) {
+                                $elements.each(function() {
+                                    extraWidth += getExtraWidth($(this));
+                                });
+                                return {
+                                    $elements: $elements,
+                                    extraWidth: extraWidth
+                                };
+                            }
+                            return false;
+                        }).filter(function (collapsible) {
```

Comment:
Instead of using such a filter, you could also rely on the `_.compact()` helper, that does the same: https://lodash.com/docs/2.4.2#compact